docs/Modern-PersistedQueries.md

```diff
@@ -0,0 +1,164 @@
+}
+```
+
+The example above writes the complete query map file to `./src/queryMaps/queryMap.desktop.queryMap.json`. You need to ensure all the directories
+leading to the `queryMap.json` file exist. Also note that the file extension has to be `.json`.
+
+### Network layer changes
+You'll need to modify your network layer fetch implementation to pass a queryId parameter in the POST body instead of a query parameter:
+
+```js
+function fetchQuery(operation, variables,) {
+  return fetch('/graphql', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    body: JSON.stringify({
+      queryId: operation.id, // NOTE: pass md5 hash to the server
+      // query: operation.text, // this is now obsolete because text is null
+      variables,
+    }),
+  }).then(response => {
+    return response.json();
+  });
+}
+```
+
+## Usage on the server
+On the server, you'll need to map the query id in the POST body to the real operation text. You can utilise the
+`complete.queryMap.json` file to do this so you'll need a copy of it on your server.
+
+For universal applications where the client and server code are in one project, this is not an issue since you can place
+the query map file in a common location accessible to both the client and the server.
+
+### Compile time push
+For applications where the client and server projects are separate, one option is to have an additional npm run script
+to push the query map at compile time to a location accessible by your server:
+
+```js
+"scripts": {
+  "push-queries": "node ./pushQueries.js",
+  "relay": "relay-compiler --src ./src --schema ./schema.graphql --persist && npm run push-queries"
+}
+```
+
+Some possibilities of what you can do in `./pushQueries.js`:
+
+* `git push` to your server repo
+
+* save the query maps to a database
+
+### Run time push
+A second more complex option is to push your query maps to the server at runtime, without the server knowing the query ids at the start.
+The client optimistically sends a query id to the server, which does not have the query map. The server then in turn requests
+for the full query text from the client so it can cache the query map for subsequent requests. This is a more complex approach
+requiring the client and server to interact to exchange the query maps.
+
+### Simple server example
+Once your server has access to the query map, you can perform the mapping. The solution varies depending on the server and
+database technologies you use, so we'll just cover the most common and basic example here.
+
+If you use `express-graphql` and have access to the query map file, you can import the `complete.queryMap.json` file directly and
+perform the matching using the `matchQueryMiddleware` from [relay-compiler-plus](https://github.com/yusinto/relay-compiler-plus).
+
+```js
+import Express from 'express';
+import expressGraphl from 'express-graphql';
```

Comment:
nit: typo! correct to expressGraph"Q"L :)

Reply:
Fixed. Thank you.